lib/t/accounts/profile.ex

```diff
@@ -92,11 +98,125 @@ defmodule T.Accounts.Profile do
     profile
     |> cast(attrs, [:story])
     |> update_change(:story, fn story -> Stickers.fix_story(story) end)
+    |> validate_story()
   end
 
-  def changeset(profile, attrs, opts \\ []) do
-    profile
-    |> essential_info_changeset(attrs, opts)
-    |> story_changeset(attrs)
+  defp validate_story(%Ecto.Changeset{changes: %{story: story}} = cs) do
+    case validate_story(story || [], _validated = [], _errors = []) do
+      {:ok, story} ->
+        force_change(cs, :story, story)
+
+      {:error, errors} when is_list(errors) ->
+        Enum.reduce(errors, cs, fn error, cs -> add_error(cs, :story, error) end)
+    end
+  end
+
+  defp validate_story(changeset), do: changeset
+
+  @spec validate_story([map], [map], [String.t()]) :: {:ok, [map]} | {:error, [String.t()]}
+  defp validate_story([page | pages], validated_story, errors) do
+    {page, new_errors} = validate_page(page)
+    validate_story(pages, [page | validated_story], new_errors ++ errors)
+  end
+
+  defp validate_story([], validated, _no_errors = []), do: {:ok, :lists.reverse(validated)}
+  defp validate_story([], _validated_story, errors), do: {:error, errors}
+
+  @spec validate_page(map) :: {map, [String.t()]}
+  defp validate_page(%{"labels" => labels} = page) do
+    {labels, errors} = validate_labels(labels || [], _validated = [], _errors = [])
+    {%{page | "labels" => labels}, errors}
```

Comment:
only labels are validated right now, other page validation can be put here in the future